releasenotes/notes/hls-with-ancillas-d6792b41dfcf4aac.yaml

```diff
@@ -0,0 +1,18 @@
+---
+features_transpiler:
+  - |
+    A new argument ``is_zero_initialized`` has been added to :func:`qiskit.compiler.transpile`,
+    :func:`.generate_preset_pass_manager`, and to :class:`~.PassManagerConfig`.
+    If set to ``True``, the qubits are assumed to be initially in the state :math:`|0\rangle`,
+    potentially allowing additional optimization opportunities for individual transpiler passes.
+  - |
+    The constructor for :class:`.HighLevelSynthesis` transpiler pass now accepts an
+    additional argument ``is_zero_initialized``. If set to ``True``, the pass assumes that the
+    qubits are initially in the state :math:`|0\rangle`. In addition, the pass keeps track of
+    clean and dirty auxiliary qubits throughout the run, and passes this information to plugins
+    via kwargs ``_num_clean_ancillas`` and ``_num_dirty_ancillas``.
```

Comment:
Is there a reason to have these with leading underscores? It seems that no other arguments are with leading underscores and the number of ancillas seem like normal arguments to me 🙂

Reply:
I was hesitating about this and changed back and forth several times myself. The original reason is to differentiate between user-specified plugin options and options set by HighLevelSynthesis pass itself. Right now I agree that the version without the underscores is better.